org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring/participant/ChangeConverter.xtend

```diff
@@ -63,19 +64,24 @@ class ChangeConverter implements IAcceptor<IEmfResourceChange> {
 	protected def dispatch void handleReplacements(IEmfResourceChange change) {
 		val outputStream = new ByteArrayOutputStream
 		tryWith(outputStream) [
+			val file = change.resource.URI.toFile
+			if (file.isReadOnly) 
+				issues.add(Severity.ERROR, '''Affected file '�file.fullPath�' is read-only''')
```

Comment:
I wonder if this should be customizable, .e.g in Eclipse you would want to use `org.eclipse.core.resources.IWorkspace.validateEdit(IFile[], Object)` instead of a plain readonly check

Reply:
Hmm, I would have expected that to check for _derived_ as well, but it doesn't. Anyway good hint!